### PR TITLE
Consolidate initial state retrieval and retrieve user email from IAP headers

### PR DESCRIPTION
**Pull request by prodonjs**
- Eliminates separate call to fetch namespaces.
- Retrieves user email address from deployments secured by IAP.

/area front-end
/area centraldashboard
/cc @avdaredevil

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/3420)

### CODE REVIEW

**Comment by prodonjs** [comment]
closes #3348 

Validate behavior in: https://kf-dev.endpoints.prodonjs-kubeflow-dev.cloud.goog

File: components/centraldashboard/app/api.ts [hunk 1]
```diff
@@ -3,72 +3,100 @@ import express from 'express';
...
+    return this.platformInfo;
+  }
+
+  /**
+   * Retrieves user information from headers.
+   * Supports:
+   *  GCP IAP (https://cloud.google.com/iap/docs/identity-howto)
+   */
+  private getUser(req: express.Request): string {
+    let email = 'anonymous@kubeflow.org';
```
Comment (avdaredevil) [inline components/centraldashboard/app/api.ts]: _**nit**_ Code depth
```suggestion
    const iap = req.header(IAP_HEADER)
    return iap? iap.slice(IAP_PREFIX.length) : 'anonymous@kubeflow.org';
```

File: components/centraldashboard/app/api.ts [hunk 2]
```diff
@@ -3,72 +3,100 @@ import express from 'express';
...
+              console.log('getting env-info');
+              const [platform, user, namespaces] = await Promise.all([
+                this.getPlatformInfo(),
+                this.getUser(req),
+                this.k8sService.getNamespaces(),
+              ]);
+              res.json({
+                platform,
+                user,
+                namespaces: namespaces.map((n) => n.metadata.name),
```
Comment (avdaredevil) [inline components/centraldashboard/app/api.ts]: Keep in mind this needs to be fresh as of 30 seconds, as per @kunmingg's suggestion. So we may need another API to keep this fresh, or use sockets

File: components/centraldashboard/app/api.ts [hunk 3]
```diff
@@ -3,72 +3,100 @@ import express from 'express';
...
+                    res.json(await this.metricsService.getPodCpuUtilization(
+                        interval));
+                    break;
+                  case 'podmem':
+                    res.json(
+                        await this.metricsService.getPodMemoryUsage(interval));
+                    break;
+                  default:
+                }
+              } else {
```
Comment (avdaredevil) [inline components/centraldashboard/app/api.ts]: **_code depth--_** Remove `else`, put a return after `switch`

File: components/centraldashboard/app/api.ts (hunk 2 above)
Comment (prodonjs) [reply components/centraldashboard/app/api.ts]: What specifically needs to be fresh, the list of namespaces? In that case we still have the dedicated namespaces endpoint to get the latest content back.

File: components/centraldashboard/app/api.ts (hunk 3 above)
Comment (prodonjs) [reply components/centraldashboard/app/api.ts]: Good call. Refactored to early return.

File: components/centraldashboard/app/api.ts (hunk 1 above)
Comment (prodonjs) [reply components/centraldashboard/app/api.ts]: I was expecting that eventually there might be additional headers that would be interrogated to support other authentication providers besides IAP.

Comment (avdaredevil) [reply components/centraldashboard/app/api.ts]: I see, sure, we can leave it as it is.

File: components/centraldashboard/app/api.ts (hunk 2 above)
Comment (avdaredevil) [reply components/centraldashboard/app/api.ts]: Yup, it was more an FYI, than an ask for change. I thought your commit message said we removed that endpoint.

**Pull request merged**